pkg/resource/podtemplatespec.go

```diff
@@ -336,13 +336,21 @@ func makePodTemplateSpec(coreClient coreset.CoreV1Interface, proxyLister configl
 		}
 	}
 
+	nodeSelectors := map[string]string{}
```

Comment:
I'm not sure how I feel about this.  The image registry has to run on linux, but we are saying that an administrator can set a different os, even if it won't run there? Seems sketchy.

Reply:
Settings `kubernetes.io/os` in this selector to anything other than `linux` is not supported today. However, if a customer decides to change its value anyway, the configuration should be broken today.

I can envision a situation when someone sets `kubernetes.io/os` to `windows` and forget about it, it works. Later we add support of this OS and his configuration suddenly has a new meaning.

Reply:
> Settings kubernetes.io/os in this selector to anything other than linux is not supported today

I don't think that is true. We are moving towards supporting Windows Containers (and that means Windows nodes) in the near future. My understanding is that this is "dev preview" in 4.4.

Reply:
@adambkaplan are you saying that someone is trying to build the image registry for Windows?

Reply:
@dmage - at this point no, just that there will be Windows workers and we need to ensure we don't schedule or workloads on them by default. If a customer sets the node selector to be `windows`, their cluster goes degraded. So 🤷

Reply:
I agree with @dmage. It is better to crash with a wrong configuration now. This way we assure that the configuration reflects user intentions once we start supporting Window hosts.

Reply:
Hey guys, any extra consideration here? I would like to get this moving ahead.